mid-assessment/q4.rb

```diff
@@ -0,0 +1,66 @@
+# 4. Design regex for mobile number, email address, name, gender(M / F) and amount
+# Personal Info module extract all the details from given text and display it.
+module PersonInfo
+  def self.mobile_number(mystr)
+    puts "Mobile Number : #{mystr.match(/\d{10}/)}"
+  end
+
+  def self.email_address(mystr)
+    puts "Email : #{mystr.match(/[\w.]+@[\w.]+/)}"
+  end
+
+  def self.name(mystr)
+    puts "Name : #{mystr.match(/[A-Z]\w+ [A-Z]\w+/)}"
+  end
+
+  def self.gender(mystr)
+    puts "Gender : #{mystr.match(/Gender - \w/)}"
+  end
+
+  def self.amount(mystr)
+    puts "Salary : #{mystr.match(/Rs.\d+/)}"
+  end
+
+  def self.details(mystr)
+    puts 'All Details'
+    name(mystr)
+    email_address(mystr)
+    mobile_number(mystr)
+    gender(mystr)
+    amount(mystr)
+  end
+end
+
+def main(mystr)
+  i = 1
+  while i.positive?
+    puts '1. Name'
+    puts '2. Email Address'
```

Comment:
Use same pattern of blank spaces for all the puts sentences

Reply:
ok